workbench-core/accounts/src/models/projects/assignUserToProjectRequest.ts

```diff
@@ -3,12 +3,12 @@
  *  SPDX-License-Identifier: Apache-2.0
  */
 
-import { z } from 'zod';
+import { resourceTypeToKey, z } from '@aws/workbench-core-base';
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const AssignUserToProjectRequestParser = z
   .object({
-    projectId: z.string(),
+    projectId: z.string().swbId(resourceTypeToKey.project.toLowerCase()).required(),
```

Comment:
If my [PR](https://github.com/aws-solutions/solution-spark-on-aws/pull/1045) merges first, you can replace this with `z.string.projId().required()`